"Can't pickle" error message changed?

[In more than one PR](https://github.com/scikit-hep/awkward/actions/runs/10495088863/job/29097093135?pr=3222), we're starting to see the following error (likely unrelated to those PRs).

```
=================================== FAILURES ===================================
__________________ test_serialise_with_nonserialisable_attrs ___________________

array_pickler = <module 'pickle' from '/opt/hostedtoolcache/Python/3.12.5/x64/lib/python3.12/pickle.py'>

    def test_serialise_with_nonserialisable_attrs(array_pickler):
        attrs = {**SOME_ATTRS, "non_transient_key": lambda: None}
        array = ak.Array([1, 2, 3], attrs=attrs)
        with pytest.raises(
            (AttributeError, array_pickler.PicklingError), match=r"Can't pickle"
        ):
>           array_pickler.loads(array_pickler.dumps(array))
E           AttributeError: Can't get local object 'test_serialise_with_nonserialisable_attrs.<locals>.<lambda>'

array      = <Array [1, 2, 3] type='3 * int64'>
array_pickler = <module 'pickle' from '/opt/hostedtoolcache/Python/3.12.5/x64/lib/python3.12/pickle.py'>
attrs      = {'foo': '!SOME',
 'non_transient_key': <function test_serialise_with_nonserialisable_attrs.<locals>.<lambda> at 0x7fac6ed2ba60>}

tests/test_2757_attrs_metadata.py:47: AttributeError

During handling of the above exception, another exception occurred:

array_pickler = <module 'pickle' from '/opt/hostedtoolcache/Python/3.12.5/x64/lib/python3.12/pickle.py'>

    def test_serialise_with_nonserialisable_attrs(array_pickler):
        attrs = {**SOME_ATTRS, "non_transient_key": lambda: None}
        array = ak.Array([1, 2, 3], attrs=attrs)
>       with pytest.raises(
            (AttributeError, array_pickler.PicklingError), match=r"Can't pickle"
        ):
E       AssertionError: Regex pattern did not match.
E        Regex: "Can't pickle"
E        Input: "Can't get local object 'test_serialise_with_nonserialisable_attrs.<locals>.<lambda>'"

array      = <Array [1, 2, 3] type='3 * int64'>
array_pickler = <module 'pickle' from '/opt/hostedtoolcache/Python/3.12.5/x64/lib/python3.12/pickle.py'>
attrs      = {'foo': '!SOME',
 'non_transient_key': <function test_serialise_with_nonserialisable_attrs.<locals>.<lambda> at 0x7fac6ed2ba60>}

tests/test_2757_attrs_metadata.py:44: AssertionError
```

It seems like the error message has changed.

The test was originally added by @agoose77 in #2757 with error message "Can't pickle local object".

https://github.com/scikit-hep/awkward/blob/e7cccc9cae0d7bc5c36ee4078d0e56a934221966/tests/test_2757_attrs_metadata.py#L41-L45

Then I changed it in #2927 to "Can't pickle":

https://github.com/scikit-hep/awkward/blob/c54d0838daf99bc396c6596d7130739ad3161625/tests/test_2757_attrs_metadata.py#L41-L47

The new message, "Can't get local object 'test_serialise_with_nonserialisable_attrs.<locals>.<lambda>'", is essentially the same thing, though I don't know why the message is changing.

@agoose77, before I arbitrarily change the expected message again (or only catch the exception types and tell ruff that we don't want to constrain the error message), does anything seem wrong here?

--- a/tests/test_2757_attrs_metadata.py
+++ b/tests/test_2757_attrs_metadata.py
@@ -42,7 +42,8 @@ def test_serialise_with_nonserialisable_attrs(array_pickler):
     attrs = {**SOME_ATTRS, "non_transient_key": lambda: None}
     array = ak.Array([1, 2, 3], attrs=attrs)
     with pytest.raises(
-        (AttributeError, array_pickler.PicklingError), match=r"Can't pickle"
+        (AttributeError, array_pickler.PicklingError),
+        match=r"(pickle|local object)",
     ):
         array_pickler.loads(array_pickler.dumps(array))
 

